antlr-kotlin-runtime/src/jsTest/kotlin/com/strumenta/antlrkotlin/runtime/IdentityHashMapTests.kt

```diff
@@ -0,0 +1,169 @@
+// Copyright 2017-present Strumenta and contributors, licensed under Apache 2.0.
+// Copyright 2024-present Strumenta and contributors, licensed under BSD 3-Clause.
+package com.strumenta.antlrkotlin.runtime
+
+import kotlin.collections.MutableMap.MutableEntry
+import kotlin.test.*
+
+class IdentityHashMapTests {
+  @Test
+  fun insertions() {
+    val imap = IdentityHashMap<Any, String>()
+    var prev = imap.put("one", "two")
+
+    assertNull(prev)
+    assertEquals(1, imap.size)
+    assertEquals("two", imap["one"])
+
+    prev = imap.put("one", "three")
+    assertEquals(1, imap.size)
+    assertEquals("two", prev)
+    assertEquals("three", imap["one"])
+
+    imap.clear()
+    assertTrue(imap.isEmpty())
+
+    val a = A("one")
+    prev = imap.put(a, "two")
+    assertNull(prev)
+
+    val b = B("one")
+    prev = imap.remove(b)
+    assertNull(prev)
+
+    prev = imap.remove(a)
+    assertEquals("two", prev)
+  }
+
+  @Test
+  fun containsValue() {
+    val imap = IdentityHashMap<String, A>()
```

Comment:
I think it would be useful to write a test using `IdentityHashMap<A, String>()` and then using different instances of `A` that are equals but are different instances as keys